modAionImpl/src/org/aion/zero/impl/AionHub.java

```diff
@@ -241,36 +240,28 @@ private void loadBlockchain() {
             if (recovered) {
                 bestBlock = this.repository.getBlockStore().getBestBlock();
 
-                // checking is the best block has changed since attempting
-                // recovery
+                // checking is the best block has changed since attempting recovery
                 if (bestBlock == null) {
                     bestBlockShifted = true;
                 } else {
-                    bestBlockShifted = !(bestBlockNumber == bestBlock.getNumber()) || // block
-                                                                                      // number
-                                                                                      // changed
-                            !(Arrays.equals(bestBlockRoot, bestBlock.getStateRoot())); // root
-                                                                                       // hash
-                                                                                       // changed
+                    bestBlockShifted = !(bestBlockNumber == bestBlock.getNumber()) || // block number changed
+                            !(Arrays.equals(bestBlockRoot, bestBlock.getStateRoot())); // root hash changed
                 }
 
                 if (bestBlockShifted) {
-                    LOG.info("Rebuilding world state SUCCEEDED. However, the best block CHANGED.");
-                    // reinitializing recovery flag in case this is the last
-                    // attempt
-                    recovered = false;
+                    LOG.info("Rebuilding world state SUCCEEDED by REVERTING to a previous block.");
                 } else {
                     LOG.info("Rebuilding world state SUCCEEDED.");
                 }
             } else {
-                LOG.info("Rebuilding world state FAILED.");
+                LOG.error("Rebuilding world state FAILED. "
+                        + "Stop the kernel (Ctrl+C) and use the command line --revert option to move back to a valid block.");
```

Comment:
How does the user know which block number he/she should given?
Also, put a command line example into this message ?

Reply:
I improved the message and I will have a tutorial on how to choose the block number when reverting.